tests/h/accounts/__init___test.py

```diff
@@ -5,140 +5,81 @@
-    """It should log the user out if they no longer exist in the db."""
-    pyramid_request.current_route_url = lambda: '/'
-    pyramid_request.session.invalidate = mock.Mock()
-    pyramid_config.testing_securitypolicy('userid')
-    get_user.return_value = None
-
-    try:
         accounts.authenticated_user(pyramid_request)
-    except Exception:
-        pass
-
-    pyramid_request.session.invalidate.assert_called_once_with()
-
 
-@authenticated_user_fixtures
-def test_authenticated_user_does_not_invalidate_session_if_not_authenticated(get_user, pyramid_config, pyramid_request):
-    """
-    If authenticated_userid is None it shouldn't invalidate the session.
+        user_service.fetch.assert_called_once_with('userid')
 
-    Even though the user with id None obviously won't exist in the db.
+    def test_invalidates_session_if_user_does_not_exist(self,
+                                                        pyramid_config,
+                                                        pyramid_request):
+        """It should log the user out if they no longer exist in the db."""
+        pyramid_request.session.invalidate = mock.Mock()
+        pyramid_config.testing_securitypolicy('userid')
 
-    This also tests that it doesn't raise a redirect in this case.
+        try:
+            accounts.authenticated_user(pyramid_request)
+        except Exception:
+            pass
 
-    """
-    pyramid_request.current_route_url = lambda: '/'
-    pyramid_request.session.invalidate = mock.Mock()
-    pyramid_config.testing_securitypolicy()
-    get_user.return_value = None
+        pyramid_request.session.invalidate.assert_called_once_with()
 
-    accounts.authenticated_user(pyramid_request)
+    def test_does_not_invalidate_session_if_not_authenticated(self,
+                                                              pyramid_config,
+                                                              pyramid_request):
+        """
+        If authenticated_userid is None it shouldn't invalidate the session.
 
-    assert not pyramid_request.session.invalidate.called
+        Even though the user with id None obviously won't exist in the db.
 
+        This also tests that it doesn't raise a redirect in this case.
 
-@authenticated_user_fixtures
-def test_authenticated_user_redirects_if_user_does_not_exist(get_user, pyramid_config, pyramid_request):
-    pyramid_request.current_route_url = lambda: '/the/page/that/I/was/on'
-    pyramid_config.testing_securitypolicy('userid')
-    get_user.return_value = None
+        """
+        pyramid_request.session.invalidate = mock.Mock()
 
-    with pytest.raises(httpexceptions.HTTPFound) as exc:
         accounts.authenticated_user(pyramid_request)
 
-    assert exc.value.location == '/the/page/that/I/was/on', (
-        'It should redirect to the same page that was requested')
+        assert not pyramid_request.session.invalidate.called
 
+    def test_redirects_if_user_does_not_exist(self,
```

Comment:
The fact that a method called `authenticated_user` can trigger an implicit redirect seemed like something of a WTF to me, but I've seen the FIXME in the implementation